src/osd/OSDMap.cc

```diff
@@ -7201,6 +7201,24 @@ void OSDMap::check_health(CephContext *cct,
 			    ss.str(), 0);
     }
   }
+  // UNEQUAL_WEIGHT
+  if (stretch_mode_enabled) {
+    vector<int> subtrees;
```

Comment:
get_subtree_of_type() is hardly used, and in all placed - the vector is an out-only
parameter. Maybe this is a good opportunity to modify its signature
to
vector<int> get_subtree_of_type(bucket)